src/index.ts

```diff
@@ -205,7 +205,7 @@ export class Upload extends Pumpify {
     this.bucket = cfg.bucket;
 
     const cacheKeyElements = [cfg.bucket, cfg.file];
-    if (cfg.generation) {
+    if (!isNaN(cfg.generation!)) {
```

Comment:
NaN is actually pretty weird. Can you do "if (typeof cfg.generation === 'number') {"?

Reply:
Updated condition.